src/smartbanner.js

```diff
@@ -106,11 +107,43 @@ export default class SmartBanner {
   }
 
   get buttonUrl() {
-    if (this.platform === 'android') {
-      return this.options.buttonUrlGoogle;
-    } else if (this.platform === 'ios') {
-      return this.options.buttonUrlApple;
+    let options = this.options;
+    let device = this.device;
+
+    if (this.ignoreIOSGlobalLink === 'iosglobal') {
+      options.buttonUrlApple = '';
+    }
+
+    if (this.ignoreAndroidGlobalLink === 'androidglobal') {
+      options.buttonUrlGoogle = '';
+    }
+
+    if (this.ignoreButtonUrlAppleIpad === 'ipad') {
+      options.buttonUrlAppleIpad = '';
+    }
+
+    if (this.ignoreButtonUrlAppleIphone === 'iphone' || this.ignoreButtonUrlAppleIpod === 'ipod') {
+      options.buttonUrlAppleIphone = '';
     }
+
+    if (this.ignoreButtonUrlGooglePhone === 'phone') {
+      options.buttonUrlGooglePhone = '';
+    }
+
+    if (this.ignoreButtonUrlGoogleTablet === 'tablet') {
+      options.buttonUrlGoogleTablet = '';
+    }
+
+    if (device === 'ipad') {
+      return options.buttonUrlAppleIpad ? options.buttonUrlAppleIpad : options.buttonUrlApple;
+    } else if (device === 'iphone' || device === 'ipod') {
+      return options.buttonUrlAppleIphone ? options.buttonUrlAppleIphone : options.buttonUrlApple;
+    } else if (device === 'phone') {
+      return options.buttonUrlGooglePhone ? options.buttonUrlGooglePhone : options.buttonUrlGoogle;
+    } else if (device === 'tablet') {
+      return options.buttonUrlGoogleTablet ? options.buttonUrlGoogleTablet : options.buttonUrlGoogle;
+    }
+
```

Comment:
This logic should be simplified, it's too long. `device` and `platform` field combination from `Detector` should become sufficient to decide what to return.